Lesson03/Lesson03/Program.cs

```diff
@@ -18,264 +18,79 @@
-
-            //DisplayNumbers(numbers, IsEven);
-            //DisplayNumbers(numbers, IsPositive);
-
-            string language = Console.ReadLine();
-
-            if (language == "uz")
-            {
-                StartApplication(WelcomeUserUzb);
-            }
-            else
-            {
-                StartApplication(WelcomeUserEn);
-            }
+            int[] number = { 2, 2, 4, 8, 10 };
+            int[] number1 = { 1, 3, 5, 9, 7 };
+            CheckNumber(number, All);
+            CheckOdd(number1, All);
+            AverageValue(number, Sum);
+            Average(number1, Aver);
         }
-
-        static void DisplayHello(string name, int age)
+        static void CheckNumber(int[] number, CheckEvenNumber checkEvenNumber)
         {
-            if (age > 17)
+            foreach (int num in number)
             {
-                Console.WriteLine($"Hello: {name}. You can take driver license");
+                if (!checkEvenNumber(num))
+                {
+                    Console.WriteLine("False");
+                    Environment.Exit(0);
+                }
             }
-            else
-            {
-                Console.WriteLine($"Hello: {name}. You cannot take driver license");
-            }
-        }
-
-        static void StartApplication(Greet greet)
-        {
-            bool isSignedIn = true;
-
-            greet(isSignedIn);
+            Console.WriteLine("True");
         }
-
-        static void WelcomeUserUzb(bool isSignedIn)
+        static bool All(int x)
         {
-            if (isSignedIn)
+            if (x % 2 == 0)
             {
-                Console.WriteLine("Qayta ko'rishdan xursandmiz!");
-            }
-            else
-            {
-                Console.WriteLine("Xush kelibsiz!");
+                return true;
             }
+            return false;
         }
-
-        static void WelcomeUserEn(bool isSignedIn)
+        static void CheckOdd(int[] number1, CHeckNumberOdd cHeckNumberOdd)
         {
-            if (isSignedIn)
+            foreach (int num in number1)
             {
-                Console.WriteLine("Welcome back!");
-            }
-            else
-            {
-                Console.WriteLine("Welcome!");
+                if (cHeckNumberOdd(num))
+                {
+                    Console.WriteLine("True");
+                    Environment.Exit(0);
+                }
             }
+            Console.WriteLine("False");
         }
-
-        static void DisplayNumbers(int[] numbers, Validator validator)
+        static void AverageValue(int[] number, AverageNumber averageNumber)
         {
-            foreach (var number in numbers)
+            int sumResult = 0;
+            foreach (int num in number)
             {
-                DisplayNumberWithColor(number, validator);
+                int result = averageNumber(num);
+                sumResult += result;
```

Comment:
int result  o'zgaruvchisi ortiqcha ish bajaryapti
sumResult ga averageNumber(num) ni to'g'ridan to'g'ri qo'shsa bo'ladi